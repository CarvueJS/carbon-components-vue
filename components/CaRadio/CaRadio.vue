<template>
  <div class="bx--form-item">
    <div class="bx--radio-button-group">
      <slot></slot>
    </div>
  </div>
</template>

<script>
  export default {
    name: 'ca-radio'
  }
</script>

<style></style>
