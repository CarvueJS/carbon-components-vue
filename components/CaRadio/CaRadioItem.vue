<template>
  <div>
    <input :id="radioID" class="bx--radio-button" type="radio" :value="value" name="radio-button" :tabindex="tabIndex" :checked="checked" :disable="disable">
    <label :for="radioID" class="bx--radio-button__label">
      <span class="bx--radio-button__appearance"></span>
      <slot></slot>
    </label>
  </div>
</template>

<script>
  export default {
    name: 'ca-radio-item',
    props: {
      checked: {
        type: Boolean,
        defalut: false,
      },
      disable: {
        type: Boolean,
        defalut: false,
      },
      value: {
        type: String,
        defalut: ''
      },
      tabIndex: {
        type: Number,
        defalut: 0,
      },
      radioID: {
        type: String,
        defalut: '',
      }
    },
  }
</script>

<style></style>
