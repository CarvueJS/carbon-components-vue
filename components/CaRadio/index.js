import CaRadio from './CaRadio.vue';
import CaRadioItem from './CaRadioItem.vue';

CaRadio.install = Vue => {
  Vue.component(CaRadio.name, CaRadio);
  Vue.component(CaRadioItem.name, CaRadioItem);
}

export default CaRadio
