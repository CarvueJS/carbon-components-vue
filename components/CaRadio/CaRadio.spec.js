import { shallow } from 'vue-test-utils';
import CaRadio from './CaRadio.vue';

describe('CaRadio', () => {
  test('is a Vue instance', () => {
    const wrapper = shallow(CaRadio);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
})
