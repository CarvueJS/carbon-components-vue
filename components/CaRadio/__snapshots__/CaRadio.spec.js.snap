// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaRadio Mathes snapshot 1`] = `
<div class="bx--form-item">
  <div class="bx--radio-button-group"></div>
</div>
`;
